File: app/schemas/content.py
from pydantic import BaseModel


class Amenity(BaseModel):
    id: str
    name: str
    category: str
    description: str | None = None
    display_order: int = 0
    active: bool = True


class SmartFeature(BaseModel):
    id: str
    name: str
    benefit_statement: str
    display_order: int = 0


class LocationPoint(BaseModel):
    id: str
    name: str
    category: str
    distance_or_travel_note: str | None = None
    latitude: float | None = None
    longitude: float | None = None
    map_url: str | None = None
    display_order: int = 0

File: src/api/content.ts
import { apiRequest } from './client';

export interface AmenityApi {
  id: string;
  name: string;
  category: string;
  description?: string | null;
  display_order: number;
  active: boolean;
}

export interface SmartFeatureApi {
  id: string;
  name: string;
  benefit_statement: string;
  display_order: number;
}

export interface LocationPointApi {
  id: string;
  name: string;
  category: string;
  distance_or_travel_note?: string | null;
  latitude?: number | null;
  longitude?: number | null;
  map_url?: string | null;
  display_order: number;
}

export const listAmenities = () => apiRequest<AmenityApi[]>('/amenities');
export const listSmartFeatures = () => apiRequest<SmartFeatureApi[]>('/smart-features');
export const listLocationPoints = () => apiRequest<LocationPointApi[]>('/location-points');
